openlabcmd/openlabcmd/zk.py

```diff
@@ -206,37 +217,117 @@ def delete_node(self, node_name):
+                service_info['node_name'] = exist_node['name']
                 result.append(service_info)
-        return sorted(result)
+        return sorted(result, key=lambda x: x['node_name'])
 
-    @_client_check_wrapper
-    def get_service(self, service_name, role=None):
-        if not role:
-            if (service_name in
-                    service.service_mapping['master_service']['necessary'] +
-                    service.service_mapping['master_service']['unnecessary']):
-                role = 'master'
-            elif (service_name in
-                  service.service_mapping['slave_service']['necessary'] +
-                  service.service_mapping['slave_service']['unnecessary']):
-                role = 'slave'
-            else:
-                exceptions.ClientError("Can't find service %s" % service_name)
+    def _get_service_role(self, service_name):
+        """Get service's node role."""
+        # NOTE(wxy): This function doesn't work for the common services like
+        # 'Mysql' and 'Zookeeper'. For the common services, cmd users should
+        # specify the role by hand. For cli user, the command  can be something
+        # like: `openlab ha service get zookeeper --role slave`. For library
+        # user, he can call the code `get_service('zookeeper', role='slave')`.
+
+        if (service_name in
+                service.service_mapping['master_service']['necessary'] +
+                service.service_mapping['master_service']['unnecessary']):
+            return 'master'
+        elif (service_name in
+              service.service_mapping['slave_service']['necessary'] +
+              service.service_mapping['slave_service']['unnecessary']):
+            return 'slave'
 
+        raise exceptions.ClientError("Can't find service %s" % service_name)
+
+    def _get_service_path_and_node(self, service_name, role):
         for exist_node in self.list_nodes():
             if exist_node['role'] == role:
                 path = '/ha/%s/%s_service/%s' % (exist_node['name'], role,
                                                  service_name)
-                result = self.client.get(path)
-                result = self._bytesToDict(result[0])
-                return result
+                return path, exist_node
         raise exceptions.ClientError("Can't find service %s" % service_name)
+
+    @_client_check_wrapper
+    def get_service(self, service_name, role=None):
+        if not role:
+            role = self._get_service_role(service_name)
+        path, srv_node = self._get_service_path_and_node(service_name, role)
+        result = self.client.get(path)
+        result = self._bytesToDict(result, include_updated_at=True)
+        result['node_name'] = srv_node['name']
+        return result
+
+    @_client_check_wrapper
+    def update_service(self, service_name, role=None, alarmed=None,
+                       restarted=None, status=None, **kwargs):
```

Comment:
didn't see any usage of the `**kwargs` argument in this function

Reply:
See L328

Reply:
oh, sorry.